controllers/machine_controller.go

```diff
@@ -598,6 +617,61 @@ func (r *MachineReconciler) shouldAdopt(m *clusterv1.Machine) bool {
 	return metav1.GetControllerOf(m) == nil && !util.HasOwner(m.OwnerReferences, clusterv1.GroupVersion.String(), []string{"Cluster"})
 }
 
+func (r *MachineReconciler) watchClusterNodes(ctx context.Context, cluster *clusterv1.Cluster) error {
+	// If there is no tracker, don't watch remote nodes
+	if r.Tracker == nil {
+		return nil
+	}
+
+	if err := r.Tracker.Watch(ctx, remote.WatchInput{
+		Name:         "machine-watchNodes",
+		Cluster:      util.ObjectKey(cluster),
+		Watcher:      r.controller,
+		Kind:         &corev1.Node{},
+		EventHandler: handler.EnqueueRequestsFromMapFunc(r.nodeToMachine),
```

Comment:
Will this conflict with the Watch we are currently configuring in `controllers/machine_controller_phases.go`? It looks like the tracker will only register a single watch for any given GVK of a resource. I suspect the tracker will likely need to be updated to do a better comparison, probably also comparing the handler (and possibly eventually the predicates given as well).

Reply:
The tracker watch is keyed based on `Name` in `WatchInput`, not GVK.

Reply:
Ah, indeed, I got mixed up with the inputs to `controllers/external/tracker.go`. My apologies.

Reply:
It's all good! 👍